### PR TITLE
Nc.news.5.patch.articles

### PR DESCRIPTION
**Pull request by ColemDev**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -67,3 +67,66 @@ describe("GET /api/articles/:article_id", () => {
...
+      .then((response) => {
+        expect(response.body.article_id).toBe(6);
+        expect(response.body.title).toBe("A");
+        expect(response.body.topic).toBe("mitch");
+        expect(response.body).toHaveProperty("author");
+        expect(response.body.author).toBe("icellusedkars");
+        expect(response.body).toHaveProperty("body");
+        expect(response.body.body).toBe("Delicious tin of cat food");
+        expect(response.body).toHaveProperty("created_at");
+        //too complicated for me to set up a program to figure out the time stamp so I'm settling for it having the proberty right now.
```
Comment (alexdevdavis) [inline __tests__/app.test.js:116]: that's ok, you can delete your comment here

File: controllers/nc_news.controller.js [hunk 2]
```diff
@@ -23,3 +24,15 @@ exports.getArticleById = (req, res, next) => {
       next(err);
     });
 };
+exports.updateArticleVotes = (req, res, next) => {
+  const { inc_votes } = req.body;
+  const { article_id } = req.params;
+
+  updateTableVotes(article_id, inc_votes)
+    .then((upDatedArticle) => {
+      res.status(200).send(upDatedArticle);
```
Comment (alexdevdavis) [inline controllers/nc_news.controller.js:33]: ideally you'll return an object with the key updatedArticle (or similar), its value being the article info

File: models/nc-news.model.js [hunk 3]
```diff
@@ -21,3 +20,14 @@ exports.selectArticleById = (selectedArticle_id) => {
...
     });
 };
+exports.updateTableVotes = (article_id, inc_votes) => {
+  return db
+    .query(
+      "UPDATE articles SET votes = votes + $1 WHERE article_id = $2 RETURNING *;",
+      [inc_votes, article_id]
+    )
+    .then(({ rows }) => {
+      console.log("models votes rows", rows);
```
Comment (alexdevdavis) [inline models/nc-news.model.js:30]: floaty log to be removed

**Review by alexdevdavis — APPROVED** [review]
good stuff, just a few comments to act on

File: __tests__/app.test.js (hunk 1 above)
Comment (ColemDev) [reply __tests__/app.test.js]: This is done.
When I've taken a comment like this on board and fixed the issue to the senior devs satisfaction. Do I click "Resolve conversation?

File: __tests__/app.test.js [hunk 4]
```diff
@@ -67,3 +67,67 @@ describe("GET /api/articles/:article_id", () => {
...
+      .patch("/api/articles/8")
+      .send(newVote)
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe(
+          "400 Bad Request: incorrect type: vote must be a number"
+        );
+      });
+  });
+});
```
Comment (k-huggs) [inline __tests__/app.test.js:133]: Need a couple more errors handled
1. status 404 /api/articles/12345459 - id does not exist
2. status 400 /api/articles/not-an-id - invalid id provided

**Comment by k-huggs** [comment]
Couple more errors to handle have a look at feedback

File: __tests__/app.test.js (hunk 4 above)
Comment (ColemDev) [reply __tests__/app.test.js:153]: Done and pushed :)

File: models/nc-news.model.js (hunk 3 above)
Comment (ColemDev) [reply models/nc-news.model.js]: All floaty logs have been removed.

File: controllers/nc-news.controller.js [hunk 5]
```diff
@@ -23,3 +24,15 @@ exports.getArticleById = (req, res, next) => {
       next(err);
     });
 };
+exports.updateArticleVotes = (req, res, next) => {
+  const { inc_votes } = req.body;
+  const { article_id } = req.params;
+
+  updateTableVotes(article_id, inc_votes)
+    .then((upDatedArticle) => {
```
Comment (alexdevdavis) [inline controllers/nc-news.controller.js:32]: updated is one word, so camelcase would be updatedArticle for this variable

**Review by alexdevdavis — APPROVED** [review]

File: controllers/nc-news.controller.js (hunk 5 above)
Comment (ColemDev) [reply controllers/nc-news.controller.js]: Sorted thank you.

**Pull request merged**